cooked-validators/src/Cooked/MockChain/Monad/Staged.hs

```diff
@@ -57,7 +57,15 @@ type InterpMockChain = MockChainT (WriterT TraceDescr [])
 --  > =~= st -> (WriterT TraceDescr []) (Either err (a, st))
 --  > =~= st -> [(Either err (a, st) , TraceDescr)]
 interpret :: StagedMockChain a -> InterpMockChain a
-interpret = mapMockChainT (mapWriterT $ flip evalStateT []) . interpretNonDet
+interpret = mapMockChainT (mapWriterT $ flip evalStateT []) . pruneSomewheres . interpretNonDet
+  where
+    -- When returning, if we are returning from a point where a /Somewhere/ modality is yet to be consumed,
+    -- return empty. If we return a, it would correspond to a trace where the modality was never applied.
+    pruneSomewheres :: (MonadPlus m) => InterpMockChainMod m a -> InterpMockChainMod m a
+    pruneSomewheres ma = do
+      result <- ma
+      ms <- lift get
+      if any isSomewhere ms then empty else return result
```

Comment:
It seems to me that `isSomewhere` is only used here, but it's defined more than a hundred lines away.

Reply:
Great catch; should go to a where clause!